src/extensions/ExtensionManager.as

```diff
@@ -342,18 +371,53 @@ public class ExtensionManager {
 	//------------------------------
 
 	public function primExtensionOp(b:Block):* {
-		var i:int = b.op.indexOf('.');
-		var extName:String = b.op.slice(0, i);
+		var unpackedOp:Array = unpackExtensionAndOp(b.op);
+		var extName:String = unpackedOp[0];
 		var ext:ScratchExtension = extensionDict[extName];
 		if (ext == null) return 0; // unknown extension
-		var primOrVarName:String = b.op.slice(i + 1);
+		var primOrVarName:String = unpackedOp[1];
 		var args:Array = [];
-		for (i = 0; i < b.args.length; i++) {
+		for (var i:int = 0; i < b.args.length; i++) {
 			args.push(app.interp.arg(b, i));
 		}
 
 		var value:*;
-		if (b.isReporter) {
+		if(b.isHat && b.isAsyncHat){
+			if(b.requestState == 0){
+				request(extName, primOrVarName, args, b);
+				app.interp.doYield();
+				return null;
+			}
+			else if(b.requestState == 2){
+				b.requestState = 0;
+				if(b.forceAsync){
+					value = b.response as Boolean;
+				}
+				else{
+					var responseObj:Object = b.response as Object;
+					args.push(responseObj);
+					if(responseObj && responseObj.hasOwnProperty('predicate')){
+						app.externalCall('ScratchExtensions.getReporter', function(v:*):void {
+							value = v;
+						}, ext.name, responseObj.predicate, args);
+					}
```

Comment:
Good point, fixed in [this commit](https://github.com/LLK/scratch-flash/pull/1032/commits/0af72a727824aa20d373071f0bb5defd66728834)